test/algorand-lib/Test/Data/Algorand/Transaction.hs

```diff
@@ -45,6 +45,9 @@ genSizedBytes gen = G.justT (sizedByteArray <$> gen (R.singleton size))
 genStateSchema :: MonadGen m => m StateSchema
 genStateSchema = StateSchema <$> G.word64 (R.linear 0 1000) <*> G.word64 (R.linear 0 1000)
 
+genOnComplete :: MonadGen m => m OnComplete
+genOnComplete = (toEnum . fromIntegral) <$> G.word64 (R.linear 0 5)
```

Comment:
I would add `Bounded` derivation and use min and max bound here

Reply:
Added Bounded and update this gen now.